apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -249,13 +282,47 @@ protected Optional<TableMetadata> mapTableResponseToTableMetadata(
     return Optional.of(builder.build());
   }
 
+  protected List<TableDataLayoutMetadata> mapTableResponseToTableDataLayoutMetadataList(
+      GetTableResponseBody shallowResponseBody) {
+    GetTableResponseBody tableResponseBody =
+        getTable(shallowResponseBody.getDatabaseId(), shallowResponseBody.getTableId());
+
+    if (tableResponseBody == null) {
+      log.error(
+          "Error while fetching metadata for table: {}.{}",
+          shallowResponseBody.getDatabaseId(),
+          shallowResponseBody.getTableCreator());
+      return Collections.emptyList();
+    }
+
+    TableDataLayoutMetadata.TableDataLayoutMetadataBuilder<?, ?> builder =
+        TableDataLayoutMetadata.builder()
+            .creator(tableResponseBody.getTableCreator())
+            .dbName(tableResponseBody.getDatabaseId())
+            .tableName(tableResponseBody.getTableId())
+            .isPrimary(
+                tableResponseBody.getTableType()
+                    == GetTableResponseBody.TableTypeEnum.PRIMARY_TABLE)
+            .isTimePartitioned(tableResponseBody.getTimePartitioning() != null)
+            .isClustered(tableResponseBody.getClustering() != null)
+            .retentionConfig(getTableRetention(tableResponseBody).orElse(null));
+    if (tableResponseBody.getCreationTime() != null) {
```

Comment:
Good catch, it seems to be legacy and probably was added because the shallow response returned by search API doesn't have most of the fields set, including this one. Since the response in this place was fetched with get API, it must have create time. Replaced with Objects.requireNonNull